File: csharp/lib/RawData/RotationData.cs
/*
 * SPDX-FileCopyrightText: 2017-2022 Sony Semiconductor Solutions Corporation
 *
 * SPDX-License-Identifier: Apache-2.0
 */

using MessagePack;

namespace SensCord
{
    /// <summary>
    /// Data for rotation attitude.
    /// </summary>
    [MessagePackObject]
    public class RotationData
    {
        /// <summary>
        /// Raw data type.
        /// </summary>
        [IgnoreMember]
        public string Type { get; } = RawDataType.Rotation;

        /// <summary>
        /// Roll angle.
        /// </summary>
        [Key("roll")]
        public float roll { get; set; }

        /// <summary>
        /// Pitch angle.
        /// </summary>
        [Key("pitch")]
        public float pitch { get; set; }

        /// <summary>
        /// Yaw angle.
        /// </summary>
        [Key("yaw")]
        public float yaw { get; set; }

        /// <summary>
        /// To string.
        /// </summary>
        /// <returns>A string that represents the current object.</returns>
        public override string ToString()
        {
            return $"{this.Type}(roll={this.roll}, pitch={this.pitch}, yaw={this.yaw})";
        }
    }
}

File: java/senscord/src/senscord_types/RotationData.java
/*
 * SPDX-FileCopyrightText: 2019 Sony Semiconductor Solutions Corporation
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package senscord_types;

import com.sun.jna.Structure;
import java.util.Arrays;
import java.util.List;

/**
 * Data for rotating posture.
 * 
 * @see "senscord::RotationData"
 */
public class RotationData extends Structure {
    public static class ByReference extends RotationData implements Structure.ByReference {
    }

    public float roll;
    public float pitch;
    public float yaw;

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("roll", "pitch", "yaw");
    }

    /**
     * Get roll angle.
     * 
     * @return roll angle
     */
    public float GetRoll() {
        return roll;
    }

    /**
     * Get pitch angle.
     * 
     * @return pitch angle
     */
    public float GetPitch() {
        return pitch;
    }

    /**
     * Get yaw angle.
     * 
     * @return yaw angle
     */
    public float GetYaw() {
        return yaw;
    }

}
